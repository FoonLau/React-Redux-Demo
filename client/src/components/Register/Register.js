import React from 'react'
import { 
  Grid,
  Row,
  Col } from 'react-bootstrap'
import RegisterForm from './RegisterForm/RegisterForm'

export default () => (
  <Grid>
    <Row>
      <Col xs={12} sm={8} smOffset={2}>
        <RegisterForm />
      </Col>
    </Row>
  </Grid>
)
