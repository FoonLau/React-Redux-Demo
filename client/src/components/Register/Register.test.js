import React from 'react';
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { fromJS } from 'immutable'
import Register from './Register';

const store = configureMockStore()({
  app: fromJS({}),
  form: {  register: {} }
})


it('should render correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Register />
    </Provider>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})