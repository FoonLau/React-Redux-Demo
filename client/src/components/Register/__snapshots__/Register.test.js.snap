// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-sm-8 col-sm-offset-2 col-xs-12"
    >
      <div
        className="register-form row"
      >
        <div
          className="col-xs-12"
        >
          <header
            className="header"
          >
            <h1
              className="text"
            >
              Register account
            </h1>
          </header>
        </div>
        <div
          className="col-xs-12"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="field"
            >
              <label
                className="field-label"
              >
                First Name
              </label>
              <div>
                <input
                  className="field-input"
                  name="firstName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="First Name"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="field"
            >
              <label
                className="field-label"
              >
                Last Name
              </label>
              <div>
                <input
                  className="field-input"
                  name="lastName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Last Name"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="field"
            >
              <label
                className="field-label"
              >
                Email
              </label>
              <div>
                <input
                  className="field-input"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Email"
                  type="email"
                  value=""
                />
              </div>
            </div>
            <div
              className="field"
            >
              <label
                className="field-label"
              >
                IBAN
              </label>
              <div>
                <input
                  className="field-input"
                  name="iban"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="IBAN"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <button
                className="pull-right btn btn-warning"
                disabled={true}
                type="submit"
              >
                Submit!
              </button>
            </div>
          </form>
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
